src/Renderers/SpanNodeRenderer.php

```diff
@@ -32,7 +37,16 @@ public function link(?string $url, string $title, array $attributes = []): strin
         $url = (string) $url;
 
         if (!$attributes) {
-            $attributes['class'] = sprintf('reference %s', 0 === strpos($url, 'http') ? 'external' : 'internal');
+            $attributes['class'] = sprintf('reference %s', $this->isExternalUrl($url) ? 'external' : 'internal');
+        }
+
+        if (
+            $this->urlChecker &&
+            $this->isExternalUrl($url) &&
+            false === strpos($url, 'http://localhost') &&
+            false === strpos($url, 'http://192.168')
+        ) {
+            $this->urlChecker->checkUrl($url);
```

Comment:
We're overriding & extending this class because there is no clean way to do it via an event, right (like you described in your comment)? If so, we should create an issue on rst-parser about that - Jon will not like this. I mean, he doesn't want us to need to override classes like this to get our job done - I think he'll want to expand the extension point.

Reply:
yes, you're absolutely right :)

Reply:
i'm gone make a PR for this

Reply:
https://github.com/doctrine/rst-parser/pull/89

Reply:
once this will be accepted, we could get rid of this object and use the new event